samples/java-protobuf-doc-snippets/src/it/java/com/example/JwtIntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+/* This code was generated by Kalix tooling.
+ * As long as this file exists it will not be re-generated.
+ * You are free to make changes to this file.
+ */
+package com.example;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import kalix.javasdk.Metadata;
+import kalix.javasdk.testkit.junit.KalixTestKitResource;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Base64;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+// Example of an integration test calling our service via the Kalix proxy
+// Run all test classes ending with "IntegrationTest" using `mvn verify -Pit`
+public class JwtIntegrationTest {
+
+  /**
+   * The test kit starts both the service container and the Kalix proxy.
+   */
+  @ClassRule
+  public static final KalixTestKitResource testKit =
+      new KalixTestKitResource(Main.createKalix());
+
+  /**
+   * Use the generated gRPC client to call the service through the Kalix proxy.
+   */
+  // tag::bearer-token-claims-test[]
+  private final JwtServiceActionClient client;
+
+  public JwtIntegrationTest() {
+    client = (JwtServiceActionClient) testKit.getGrpcClient(CounterService.class); // <1>
```

Comment:
This bit feels a bit weird.. This is due to the fact that we need to be able to inject headers and that is available otherwise. Is this worth a different alternative?

Reply:
I think that is good enough